src/dune_pkg/opam_solver.ml

```diff
@@ -573,15 +601,25 @@ let opam_package_to_lock_file_pkg
         version_by_package_name
         opam_file.depends
     with
-    | Ok dep_package_names ->
-      List.map dep_package_names ~f:(fun package_name -> Loc.none, package_name)
+    | Ok dep_package_names -> Package_name.Set.of_list dep_package_names
     | Error (`Formula_could_not_be_satisfied hints) ->
       Code_error.raise
         "Dependencies of package can't be satisfied from packages in solution"
         [ "package", Dyn.string (OpamFile.OPAM.package opam_file |> OpamPackage.to_string)
         ; "hints", Dyn.list Resolve_opam_formula.Unsatisfied_formula_hint.to_dyn hints
         ]
   in
+  let depopts = available_depopts solver_env version_by_package_name opam_file in
+  print_endline
+    (sprintf
+       "included depopts from package %S: %s"
+       (Package_name.to_string name)
+       (Package_name.Set.to_dyn depopts |> Dyn.to_string));
```

Comment:
Leftover debug print (I guess you're aware of it, just to not forget it)

Reply:
ah good catch